### PR TITLE
Change marker-output and sourceFrame-String

### PR DESCRIPTION
**Pull request by fzachert**
Hi,

I made some changes, to make the interface of this detection-task more similar to the aruco-detector-task.
1. The output is now a single RigidbodyState and a vector of all detected RigidBodyStates. In the old version, the output was simply the Pose of the first detected Marker, even there were multiple marker in the image.
2. The output-pose of the marker is now the pose of the marker with respect to the camera. In the old version, the output was inversed and was the camera-pose with respect to the marker (which violates the port-documentation in the orogen-file)
3. I added the marker-ID to the sourceFrame-String of the RigidBodyState. The string is now "apriltag_id_X_frame", where X is the ID of the marker. This is similar to the output of the aruco-Task. In the old version, there was only a constant string with no informations about the marker-ID.

### CODE REVIEW

File: apriltags.orogen [hunk 1]
```diff
@@ -46,10 +46,11 @@ task_context "Task" do
...
     output_port("output_image", ro_ptr("/base/samples/frame/Frame"))
         .doc 'Image that shows the detected markers'
-#    output_port("marker_poses", "std/vector<base::samples::RigidBodyState>")
-#        .doc 'It contains the calculated marker poses in respect to camera frame of all visible markers'
-    output_port("marker_poses", "base::samples::RigidBodyState")
+    output_port("marker_poses", "std/vector<base::samples::RigidBodyState>")
         .doc 'It contains the calculated marker poses in respect to camera frame of all visible markers'
+#    output_port("marker_poses", "base::samples::RigidBodyState")
+#        .doc 'It contains the calculated marker poses in respect to camera frame of all visible markers'
+    output_port("marker2cam", "base::samples::RigidBodyState")
```
Comment (dbcesar) [inline apriltags.orogen]: Please add a short description of this output port and you can delete the commented lines.

File: tasks/Task.cpp [hunk 2]
```diff
@@ -234,7 +234,17 @@ void Task::updateHook()
...
-				_marker_poses.write(rbs_vector[0]);
+				base::samples::RigidBodyState rbs_marker2cam = rbs_vector[0];
+				rbs_marker2cam.setTransform(  rbs_marker2cam.getTransform().inverse() );
+				std::vector<base::samples::RigidBodyState> marker_poses;
+				for(unsigned i = 0; i < rbs_vector.size(); i++)
+                {
+                    base::samples::RigidBodyState marker2cam = rbs_vector[i];
+                    marker2cam.setTransform(  rbs_vector[i].getTransform().inverse() );
+                    marker_poses.push_back(marker2cam);
+                }
```
Comment (dbcesar) [inline tasks/Task.cpp]: Please check the indentation here.

**Comment by dbcesar** [comment]
Hi @fzachert , thank you for the improvements, all of them are quite relevant. Please check the comments above before I merge it.  Regards,

**Comment by fzachert** [comment]
Okay, I fixed the indentation and did some documentation for the new output-port (which has now a more descriptive name).
I also removed the sourceFrame-property, because it is now unused (The source-Frame-string is now created by the marker-ID)

**Comment by dbcesar** [comment]
Merged! Thanks!

**Pull request merged**